lib/InvalidNameException.php

```diff
@@ -0,0 +1,6 @@
+<?php
+
+namespace Amp\Dns;
+
+class InvalidNameException extends \Exception {
```

Comment:
I think it should extend `ResolutionException`, even if `ResolutionException` is badly named, because it should be `DnsException`.